### PR TITLE
feat: add distributable package

### PR DESCRIPTION
**Pull request by karolsojko**
I removed the `dist` folder which will be generated with every build on `004` branch now and published as a package on Github Packages.

Added the `nvmrc` setup with a lock on latest stable `12.19.0` for node

once this is merged `snjs` will be possible to use an npm package like this:
```
// package.json
...
"@standardnotes/snjs": "^2.0.4",
...
```

### CODE REVIEW

**Review by baptiste-grob — CHANGES_REQUESTED** [review]
SNJS should be install-able without needing to create a GitHub token, otherwise anyone wanting to contribute to downstream clients (web/mobile/desktop) will have to jump through an extra hoop to run/edit the projects

File: package.json [hunk 1]
```diff
@@ -1,8 +1,14 @@
 {
-  "name": "snjs",
-  "version": "2.0.3",
+  "name": "@standardnotes/snjs",
+  "version": "2.0.4",
+  "engines": {
```
Comment (baptiste-grob) [inline package.json:4]: What does the engines key do?

Comment (karolsojko) [reply package.json:4]: https://docs.npmjs.com/cli/v6/configuring-npm/package-json#engines ;)

**Comment by karolsojko** [comment]
> SNJS should be install-able without needing to create a GitHub token, otherwise anyone wanting to contribute to downstream clients (web/mobile/desktop) will have to jump through an extra hoop to run/edit the projects

Will ammend with creating an npm registry package instead of a Github package

**Comment by karolsojko** [comment]
@baptiste-grob the package will be now published to NPM registry

File: package.json (hunk 1 above)
Comment (baptiste-grob) [reply package.json:4]: Hah, fair enough, sorry~ in that case I assume this is strictly used to help one build with a correct version of node, in which case maybe `>=12.19.0 <14.x` would likely be more correct? I'm running node 14.14.0 and building without issue for example

**Review by baptiste-grob — APPROVED** [review]
Awesome 👍

**Pull request merged**